http-router.js

```diff
@@ -351,7 +352,7 @@ function match(req, routes, i) {
         var j
         for (j = 1, len = captures.length; j < len; ++j) {
           var key = keys[j-1],
-            val = typeof captures[j] === 'string' ? decodeURIComponent(captures[j]) : captures[j];
+            val = typeof captures[j] === 'string' ? sanitizeHtml(decodeURIComponent(captures[j])) : captures[j];
```

Comment:
Front end issue aside I think we'd want a 405 error or something to respond with the ID is invalid as opposed to converting it to an acceptable value since it's important these values are exact.

I might also be concerned about the perf cost of this.